src/mumble/GKeys.cpp

```diff
@@ -0,0 +1,172 @@
+ * no parameters and does not return anything.
+ *
+ * You can poll for button states with the DLL using the functions
+ * "LogiGkeyIsMouseButtonPressed" and "LogiGkeyIsKeyboardGkeyPressed".
+ *
+ * The function "LogiGkeyIsMouseButtonPressed" takes a single int parameter, a
+ * button number. Mouse button numbers run from 6 up to and including 20. The
+ * function returns a BOOL that is true if the button is pressed, and false if
+ * not.
+ *
+ * The function "LogiGkeyIsKeyboardGkeyPressed" takes two int parameters, a
+ * button number and a mode number. Keyboard button numbers run from 1 up to
+ * and including 29. The mode number can 1, 2 or 3. The mode checks the button
+ * state in a specific mode. Typically, one queries all buttons for all modes,
+ * so one ends up with 29*3 calls to the function. The function returns a BOOL
+ * that is true if the button in the given mode is pressed, and false if not.
+ *
+ * There are also two functions, "LogiGkeyGetMouseButtonString" and
+ * "LogiGkeyGetKeyboardGkeyString". They take the same parameters as the
+ * polling functions above, but do not check whether the button is pressed or
+ * not. Instead, they return the name of the button being queried as a pointer
+ * to a NUL-terminated array of wchar_t's. Presumably, the pointer will be
+ * NULL if the name cannot be retrieved or translated.
+*/
+
+/* USAGE
+ * In order to use the gkeys on a logitech keyboard, any user must have the
+ * Logitech Gaming Software version 8.55+ installed on their computer. Then
+ * (re)start mumble. When mumble initializes the library, the LGS (Logitech
+ * Gaming Software) will create a profile called "mumble", featuring the mumble
+ * icon. In LGS, right click this icon, and select either "Set as Default" or
+ * "Set as Persistent". (See "What are persistent and default profiles?" in the
+ * LGS help by clicking on the "?" icon on the LGS window, or at
+ * http://www.logitech.com/assets/51813/3/lgs-guide.pdf). If mumble is not set
+ * as the default or persistent profile, then your keys will not be active
+ * unless mumble is the active window.
+ */
+
+#include "mumble_pch.hpp"
+
+#include "GKeys.h"
+
+#ifdef Q_CC_GNU
+#define RESOLVE(var) { var = reinterpret_cast<__typeof__(var)>(qlLogiGkey.resolve(#var)); bValid = bValid && (var != NULL); }
+#else
+#define RESOLVE(var) { * reinterpret_cast<void **>(&var) = static_cast<void *>(qlLogiGkey.resolve(#var)); bValid = bValid && (var != NULL); }
+#endif
+
+const QUuid GKeyLibrary::quMouse = QUuid(QString::fromLatin1(GKEY_MOUSE_GUID));
+const QUuid GKeyLibrary::quKeyboard = QUuid(QString::fromLatin1(GKEY_KEYBOARD_GUID));
+
+GKeyLibrary::GKeyLibrary()
+{
+	QStringList alternatives;
+
+	HKEY key = NULL;
+	DWORD type = 0;
+	LPSTR libLocation[255];
+	DWORD len = 255;
+	LONG errOpen = RegOpenKeyExA(GKEY_LOGITECH_DLL_REG_HKEY, GKEY_LOGITECH_DLL_REG_PATH, NULL, KEY_READ, &key);
+	if (errOpen == ERROR_SUCCESS) {
+		// XXX: Should the unicode variants be used instead?

```

Comment:
Yes, we simply use RegOpenKeyEx, RegQueryValueEx, etc. in most places in the rest of the codebase.